opentelemetry-sdk/benches/log_processor.rs

```diff
@@ -0,0 +1,171 @@
+use std::{sync::{Arc, Mutex}, thread::sleep, time::SystemTime};
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use opentelemetry::logs::{
+    LogRecord as _, LogResult, Logger as _, LoggerProvider as _, Severity,
+};
+use opentelemetry_sdk::{export::logs::LogData, logs::{LogProcessor, LoggerProvider}};
+
+// Run this benchmark with:
+// cargo bench --bench log_processor
+
+#[derive(Debug)]
+struct NoopProcessor;
+
+impl LogProcessor for NoopProcessor {
+    fn emit(&self, _data: &mut LogData) {}
+
+    fn force_flush(&self) -> LogResult<()> {
+        Ok(())
+    }
+
+    fn shutdown(&self) -> LogResult<()> {
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct CloningProcessor;
+
+impl LogProcessor for CloningProcessor {
+    fn emit(&self, data: &mut LogData) {
+        let _data_cloned = data.clone();
```

Comment:
Not sure if the compiler will optimize out the clone. Should we add just to ensure it doesn't:
```rust
        black_box(_data_cloned); 
```

Reply:
Shared the numbers in the top of file. The numbers from benchmark show this has much higher cost, indicating that the clone is occurring and not optimized away. Okay to merge without the black_box?